common/mariadb/templates/backup-v2-deployment.yaml

```diff
@@ -25,6 +25,9 @@ spec:
 {{- if .Values.metrics.enabled }}
         prometheus.io/scrape: "true"
         prometheus.io/targets: {{ required ".Values.alerts.prometheus missing" .Values.alerts.prometheus | quote }}
+{{- end }}
+{{- if or (and (hasKey $.Values "global") (and (hasKey $.Values.global "linkerd_enabled") $.Values.global.linkerd_enabled) (and (hasKey $.Values.global "linkerd_requested") $.Values.global.linkerd_requested)) $.Values.linkerd.backup.enabled }}
```

Comment:
I would suggest to add to the `values.yaml` a `global: {}` to ensure that the global dict always exists, it will get either merged or overwritten by the provided values file.

Combined with the default of `get` you can avoid checking the existence of keys:
```
and (get $.Values.global "linkerd_enabled") (get $.Values.global "linkerd_requested") $.Values.linkerd.backup.enabled
```

Reply:
You don't even need `get`. A single key lookup will just yield `null` by default, so just writing `.Values.global.linkerd_enabled` is completely safe (and will behave as false in boolean expressions) if `.Values.global` exists. There is really no need at all to deviate from the expressions in the linkerd-support guide, and doing so will just make it harder for me to find these instances again later on.